### PR TITLE
CA-290891: Add quicktest for VDI export & import

### PR DESCRIPTION
**Pull request by gaborigloi**
Previously VDI export wasn't covered by the quicktests.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml [hunk 1]
```diff
@@ -110,31 +110,50 @@ let checksum rpc session_id vdi =
       Digest.to_hex (Digest.file path)
     )
 
-let check_vdi_unchanged session_id ~prepare_vdi sr_info =
+let check_vdi_unchanged session_id ~prepare_vdi ~vdi_op sr_info =
   let rpc = !Quicktest_common.rpc in
   let sR = sr_info.Storage_test.sr in
-  let vdi = Client.Client.VDI.create ~rpc ~session_id ~name_label:"" ~name_description:"" ~sR ~virtual_size:4194304L ~_type:`user ~sharable:false ~read_only:false ~other_config:[] ~xenstore_data:[] ~sm_config:[] ~tags:[] in
-  Storage_test.VDI.with_destroyed session_id vdi (fun () ->
+  Storage_test.VDI.with_new session_id sR (fun vdi ->
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml]: `with_new` does the same as the VDI.create above, good.

File: ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml [hunk 2]
```diff
@@ -110,31 +110,50 @@ let checksum rpc session_id vdi =
...
       prepare_vdi vdi;
       let checksum_original = checksum rpc session_id vdi in
-      let copy = Client.Client.VDI.copy ~rpc ~session_id
-          ~vdi
-          ~base_vdi:API.Ref.null
-          ~into_vdi:API.Ref.null
-          ~sr:sR
-      in
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml]: this got moved to copy_vdi below

File: ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml [hunk 3]
```diff
@@ -110,31 +110,50 @@ let checksum rpc session_id vdi =
...
-      Storage_test.VDI.with_destroyed session_id copy (fun () ->
-          let checksum_copy = checksum rpc session_id vdi in
+      let new_vdi = vdi_op rpc session_id sR vdi in
+      Storage_test.VDI.with_destroyed session_id new_vdi (fun () ->
+          let checksum_copy = checksum rpc session_id new_vdi in
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml]: Was this buggy previously? Was it checksumming the same VDI and comparing it to itself? Nice spot.

Comment (gaborigloi) [reply ocaml/quicktest/quicktest_vdi_ops_data_integrity.ml]: Yep, it was, it was checking that `a = a` :flushed:

**Comment by gaborigloi** [comment]
If we merge before https://github.com/xapi-project/vhd-tool/pull/64, the BVT will break :smile:

**Comment by edwintorok** [comment]
The dependency of this PR got merged, so we can merge this now.

**Pull request merged**